### PR TITLE
Remove auth bloc

### PR DESCRIPTION
**Pull request by lockieRichter**
Removes the auth bloc and replaces widgets using it with providers.
Updates analysis options to have a more opinionated set of lints.

### CODE REVIEW

File: ios/Runner/Info.plist [hunk 1]
```diff
@@ -60,6 +60,8 @@
 		</array>
 		<!-- End of the Google Sign-in Section -->
 		<key>UIApplicationSupportsIndirectInputEvents</key>
-	<true/>
-</dict>
+		<true />
+		<key>FLTEnableImpeller</key>
+		<false />
```
Comment (shockbytes) [inline ios/Runner/Info.plist]: Why do you disable `Impeller` for iOS?

File: lib/main.dart [hunk 2]
```diff
@@ -64,26 +67,24 @@ class DanteXApp extends ConsumerWidget {
...
+        data: (user) {
+          if (user != null) {
+            return const MainPage();
           }
+          return const LoginPage();
+        },
+        error: (error, stackTrace) {
+          return null;
+        },
+        loading: () {
```
Comment (shockbytes) [inline lib/main.dart]: This will clash with my changes with the `go_router`.

File: lib/src/ui/profile/email_bottom_sheet.dart [hunk 3]
```diff
@@ -279,14 +266,13 @@ class EmailBottomSheetState extends ConsumerState<EmailBottomSheet> {
 
 enum LoginPhase {
   email,
-  passwordExistingUser,
```
Comment (shockbytes) [inline lib/src/ui/profile/email_bottom_sheet.dart:282]: Why did you remove this state?

File: lib/main.dart [hunk 4]
```diff
@@ -17,7 +17,7 @@ import 'package:go_router/go_router.dart';
 import 'package:google_fonts/google_fonts.dart';
 
 void main() async {
-  runZonedGuarded(
+  await runZonedGuarded(
```
Comment (shockbytes) [inline lib/main.dart:22]: Do you really need to await this future?

File: lib/main.dart [hunk 5]
```diff
@@ -128,11 +131,13 @@ final GoRouter _router = GoRouter(
...
           path: DanteRoute.settings.url,
-          builder: (BuildContext context, GoRouterState state) => const SettingsPage(),
+          builder: (BuildContext context, GoRouterState state) =>
+              const SettingsPage(),
         ),
         GoRoute(
           path: DanteRoute.profile.url,
-          builder: (BuildContext context, GoRouterState state) => const ProfilePage(),
+          builder: (BuildContext context, GoRouterState state) =>
+              const ProfilePage(),
```
Comment (shockbytes) [inline lib/main.dart:153]: Looks like we have different line lengths. What is yours?

File: lib/src/data/authentication/authentication_repository.dart [hunk 6]
```diff
@@ -1,6 +1,8 @@
 import 'package:dantex/src/data/authentication/entity/dante_user.dart';
 
 abstract class AuthenticationRepository {
+  Stream get authStateChanges;
```
Comment (shockbytes) [inline lib/src/data/authentication/authentication_repository.dart]: Please provide the type of the Stream. I think it should be `Stream<User?>`.

File: lib/src/ui/boot_page.dart [hunk 7]
```diff
@@ -13,22 +12,15 @@ class BootPage extends ConsumerStatefulWidget {
...
-    super.initState();
-  }
-
   @override
   Widget build(BuildContext context) {
+    ref.watch(userProvider).whenData((user) {
+      final String navigationUrl = user != null
+          ? DanteRoute.dashboard.navigationUrl
+          : DanteRoute.login.navigationUrl;
+      context.pushReplacement(navigationUrl);
```
Comment (shockbytes) [inline lib/src/ui/boot_page.dart]: Nice.

**Review by shockbytes — COMMENTED** [review]
Only left a couple of questions.

File: ios/Runner/Info.plist (hunk 1 above)
Comment (lockieRichter) [reply ios/Runner/Info.plist]: ah it was busted for the iOS simulator, I'll remove this - wasn't meant to commit

File: lib/main.dart (hunk 2 above)
Comment (lockieRichter) [reply lib/main.dart]: yeah have rebased now for that

File: lib/src/ui/profile/email_bottom_sheet.dart (hunk 3 above)
Comment (lockieRichter) [reply lib/src/ui/profile/email_bottom_sheet.dart:282]: I've moved this whole flow into a separate page, rather than a bottom sheet as this feels more web friendly rather than a bottom sheet.

File: lib/main.dart (hunk 4 above)
Comment (lockieRichter) [reply lib/main.dart:22]: Not strictly, but one of the lints I've enabled will make sure you await all futures (unless you wrap it in an `unawaited` block), in my experience it's better to handle futures like this - otherwise it's too easy to get race conditions from not awaiting futures

File: lib/main.dart (hunk 5 above)
Comment (lockieRichter) [reply lib/main.dart:153]: whatever the default is for flutter/dart which I think is 80

File: lib/src/data/authentication/authentication_repository.dart (hunk 6 above)
Comment (lockieRichter) [reply lib/src/data/authentication/authentication_repository.dart]: If I do that though I need to import the `firebase` package, which would make this repo no longer truly abstract?

Comment (shockbytes) [reply lib/src/data/authentication/authentication_repository.dart]: In the old app I circumvented this by introducing the [DanteUser](https://github.com/shockbytes/Dante/blob/8b5b1043adac8360226628fac4114203605c1965/core/src/main/java/at/shockbytes/dante/core/login/DanteUser.kt#L12) which simply abstracts that. I would recommend doing this here, so we are theoretically not dependent on Firebase for authentication.

File: lib/main.dart (hunk 5 above)
Comment (shockbytes) [reply lib/main.dart:153]: I'll change it to 80 too.

**Review by shockbytes — APPROVED** [review]
Looks good. Just please introduce the `DanteUser` class when exposing user data to other layers.

File: lib/src/data/authentication/authentication_repository.dart (hunk 6 above)
Comment (lockieRichter) [reply lib/src/data/authentication/authentication_repository.dart]: Good idea, all done

**Review by shockbytes — APPROVED** [review]
Nicely done
